Objective-C/CBLFullTextIndexConfiguration.m

```diff
@@ -0,0 +1,46 @@
+//
+//  CBLFullTextIndexConfiguration.m
+//  CouchbaseLite
+//
+//  Copyright (c) 2021 Couchbase, Inc All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import "CBLFullTextIndexConfiguration.h"
+#import "CBLIndexConfiguration+Internal.h"
+
+@implementation CBLFullTextIndexConfiguration
+
+@synthesize ignoreAccents=_ignoreAccents, language=_language;
+
+- (instancetype) initWithExpression: (NSArray<NSString*>*)expressions
+                      ignoreAccents: (BOOL)ignoreAccents
+                           language: (NSString* __nullable)language {
+    self = [super initWithIndexType: kC4FullTextIndex expressions: expressions];
+    if (self) {
+        _ignoreAccents = ignoreAccents;
+        _language = language ?: [[NSLocale currentLocale] objectForKey: NSLocaleLanguageCode];
```

Comment:
I think we shouldn't set the language with the current locale here so the language property set by users will not be changed even the null value is set. We can do this logic in the indexOptions.